awscli/s3transfer/copies.py

```diff
@@ -226,6 +226,8 @@ def _submit_multipart_request(
                 num_parts,
                 transfer_future.meta.size,
             )
+            # Get the checksum algorithm of the multipart request.
+            checksum_algorithm = call_args.extra_args.get("ChecksumAlgorithm")
```

Comment:
non-blocking: In the future, we should consider mapping the results of [`HeadObject`](https://github.com/aws/aws-cli/blob/9419d473fdb7f9017ff1c935da5b5cd8834ce86f/awscli/s3transfer/copies.py#L109-L113) and using that to automatically pass `ChecksumAlgorithm`. I'm mainly suggesting this because the low-level [`CopyObject` API](https://docs.aws.amazon.com/AmazonS3/latest/API/API_CopyObject.html) states:
> When you copy an object, if the source object has a checksum, that checksum value will be copied to the new object by default.

We can potentially add it in a separate PR since we want to stay aligned with the `s3transfer` package. It's also not totally straightforward to add it to [this mapping](https://github.com/aws/aws-cli/blob/9419d473fdb7f9017ff1c935da5b5cd8834ce86f/awscli/s3transfer/copies.py#L33) since `HeadObject` will return something like `ChecksumCRC32` instead of `ChecksumAlgorithm`.

Reply:
Thanks for the feedback. Agreed with your points for potentially adding it in a separate PR.